tests/js/common/shell/test-kill-queries.js

```diff
@@ -0,0 +1,328 @@
+  testCases.push(createTestCaseEntry("QueryStreamCursor::directKillAfterTrxSetup", false, "on", "on"));
+
+  // On stream the dump happens during process of the query, so it can be killed
+  if (isServer) { // shell_server
+    testCases.push(createTestCaseEntry("QueryCursor::directKillBeforeQueryIsGettingDumpedSynced", false, "on", "on"));
+    testCases.push(createTestCaseEntry("QueryCursor::directKillAfterQueryIsGettingDumpedSynced", false, "on", "off"));
+  } else { // shell_client
+    testCases.push(createTestCaseEntry("QueryCursor::directKillBeforeQueryIsGettingDumped", false, "on", "on"));
+    if (internal.isCluster()) {
+      // In cluster we can return WAITING. This will result in wakeup that will find a killed to report
+      testCases.push(createTestCaseEntry("QueryCursor::directKillAfterQueryIsGettingDumped", false, "on", "on"));
+    } else {
+      // In single server we cannot return WAITING, so once dump is finished everything is done.
+      testCases.push(createTestCaseEntry("QueryCursor::directKillAfterQueryIsGettingDumped", false, "on", "off"));
+    }
+  }
+
+  /*
+   * Non-Stream
+   */
+  // On non stream the (dump) handleQueryResult happens after query is fully processed, so it cannot be killed anymore.
+  if (isServer) { // shell_server
+    testCases.push(createTestCaseEntry("Query::executeV8directKillBeforeQueryResultIsGettingHandled", false, "off", "off"));
+    testCases.push(createTestCaseEntry("Query::executeV8directKillAfterQueryResultIsGettingHandled", false, "off", "off"));
+  } else { // shell_client
+    testCases.push(createTestCaseEntry("RestCursorHandler::directKillBeforeQueryResultIsGettingHandled", false, "off", "off"));
+  }
+
+  /*
+   * Execution in default & stream
+   */
+  testCases.push(createTestCaseEntry("ExecutionEngine::directKillBeforeAQLQueryExecute", false, 'both', "on"));
+  if (internal.isCluster()) {
+    // We wait after first Execute. SO we are not done and can kill the query.
+    testCases.push(createTestCaseEntry("ExecutionEngine::directKillAfterAQLQueryExecute", false, 'both', "on"));
+  } else {
+    // NO waiting, the query is done after a single Execute run
+    // With > 1000 results this should return an error
+    testCases.push(createTestCaseEntry("ExecutionEngine::directKillAfterAQLQueryExecute", false, 'both', "off"));
+  }
+  testCases.push(createTestCaseEntry("Query::directKillBeforeQueryWillBeFinalized", false, 'both', "both"));
+  testCases.push(createTestCaseEntry("Query::directKillAfterQueryWillBeFinalized", false, 'both', "both"));
+  testCases.push(createTestCaseEntry("Query::directKillAfterDBServerFinishRequests", true, 'both', "both"));
+
+  const testSuite = {
+    setUp: function () {
+      db._useDatabase("_system");
+      db._createDatabase(databaseName);
+      db._useDatabase(databaseName);
+
+      db._create(collectionName, {numberOfShards: 4});
+    },
+
+    tearDown: function () {
+      console.warn("CALLING TEAR DOWN, DROPPING DB");
```

Comment:
Calling `tearDown` is expected behavior. Shall we remove the `console.warn(...)` here?